### ISSUE TITLE
TCNForecaster should add validation set option for fit method

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/analytics-zoo#3743]
Now we use training data as the validation set.
